packages/jest-cli/src/runJest.js

```diff
@@ -131,12 +184,47 @@ const runJest = async (
     return onComplete && onComplete(runResults);
   };
 
-  const data = await source.getTestPaths(patternInfo);
-  processTests(data);
-  const sequencer = new TestSequencer(hasteContext);
-  const tests = sequencer.sort(data.paths);
-  const results = await runTests(tests);
-  sequencer.cacheResults(tests, results);
+  const allTests = testRunData
+    .reduce((tests, testRun) => tests.concat(testRun.tests), [])
+    .sort((a: Test, b: Test) => {
+      if (a.duration != null && b.duration != null) {
+        return a.duration < b.duration ? 1 : -1;
+      }
+      return a.duration == null ? 1 : 0;
+    });
+
+  if (!allTests.length) {
+    const localConsole = new Console(pipe, pipe);
+    localConsole.log(getNoTestsFoundMessage(testRunData, pattern));
+  }
+
+  if (
+    allTests.length === 1 &&
+    context.config.silent !== true &&
+    context.config.verbose !== false
+  ) {
+    contexts.forEach(
+      context =>
+        context.config = Object.assign({}, context.config, {verbose: true}),
+    );
+  }
+
+  const results = await new TestRunner(
+    context.config,
+    {
+      getTestSummary: () =>
```

Comment:
this looks like a leaky abstraction of SummaryReporter, is it possible to move it down the hierarchy and pass needed data?